### DISCUSSION TITLE
Table Column Actions with Forms

### DISCUSSION THREAD

**Discussion by craigkuhns** [filamentphp/filament#2916] [Ideas]
With the recent changes to how actions work  I think it would be great to be able to pass an instance of Table\Actions\Action to the action method on a table column. Currently it's possible to run some code when a table column is clicked by utilizing the action method on the columns like so:

```
Filament\Tables\Columns\BooleanColumn::make('enabled')
    ->action(fn(Model $record) => $model->update(['enabled' => !$model->enabled)
```

This works well for toggling properties of a model or for doing things where no user interaction is required. I'm running into instances where it would be great to open up a modal with a form like we can on a normal table action. For instance, I have a table with a lot of columns and the user has to scroll the table to view the edit action. For certain columns I'd like to be able to have them click the column, open a modal form to let them edit a value, and then save that value to the database. 

I'm doing this now, but it's very hacky. I've created an InvisibleAction class that extends Action and overrides the toHtml method to return an empty string so that the action doesn't show up in the UI. I then add that InvisibleAction to the tables actions like you would a normal Action. On the actual column I do this:

```
Filament\Tables\Columns\TextColumn::make('name')
    ->action(fn(Model $record, Component $livewire) => $livewire->mountTableAction('invisible-action-name', str($record->id))
``` 

That works but ideally I'd like to be able to write code that looks like this:

```
Filament\Tables\Columns\TextColumn::make('name')
    ->action(
        Filament\Tables\Actions\Action::make('edit_name')
            ->form(fn(Model $record) => [                
                Filament\Forms\Components\TextInput::make('name')
                    ->required()
                    ->default($record->name)
            ])
            ->action(fn(Model $record, array $data) => $record->update(['name' => $data['name']))
            ->visible(fn() => auth()->user()->can('edit_name'))
    )
```

**Comment by studiowizjo** [comment]
I struggled with the same and made [this PR](https://github.com/filamentphp/filament/pull/2769#issue-1271177168), to make this possible.

**Reply by danharrin** [reply]
AFAIK this is different to what @craigkuhns is requesting

**Reply by studiowizjo** [reply]
He needs table column which is able to open custom action on click. 

In linked PR is exact example how to do it if there is possibility to disable default URL:

```php
TextColumn::make('is_paid')
    ->disableUrl()
    ->extraAttributes(function (Invoice $record) {
        return [
            'wire:click' => 'mountTableAction("settle", ' . $record->getKey() . ')',
            'class' => 'cursor-pointer',
        ];
    })
```

**Reply by danharrin** [reply]
He wants to be able to pass an Action instance into the column directly though

**Comment by danharrin** [comment]
I'll add this to the roadmap 👍

**Reply by craigkuhns** [reply]
Thanks, and thanks for all of your hard work on Filament.

**Comment by danharrin** [comment]
This feature has now been implemented in #3937, thanks for your patience

**Reply by kurucu** [reply]
Literally just found a need for this. You guys are killing it.
